### ISSUE TITLE
Simplify the process of creating release notes

### ISSUE THREAD

**Issue by rok-cesnovar** [stan-dev/math#1729]
## Description
[As previously discussed](https://discourse.mc-stan.org/t/list-of-stanc3-new-reserved-keywords/12948/20) I propose we start using the RELEASE-NOTES file to update the added features and bugfixes on PRs. This will greatly simplify the burden of creating release notes. It will also make the release notes more accurate and anyone wanting to use the develop version will be aware of recent changes.

We should use the existing RELEASE-NOTES.txt file and change it to the .md file extension.

I added short notes for everything that was merged up to this release. See https://github.com/stan-dev/math/blob/2aa77bac0aba027100dc33ec4a179b2e611ce095/RELEASE-NOTES.md

I will open a PR once https://github.com/stan-dev/math/pull/1724 is merged to make this easier to read.

I propose three sections: 
- features
- bugfixes
- developer-aimed features (testing, refactors that dont change the interface)

I think a change in RELESE-NOTES.md should become mandatory. Offcourse there are expections where it doesnt make sense to enforce this. Cases where it doesnt make sense to enforce this: 
- when a feature is split in multiple PRs
- bugfix for something that was added since the last release
- CI related PRs

#### Current Version:
v3.1.0

**Comment by bob-carpenter** [comment]
I'm in favor of doing this.

Is there something to be done for this issue?

**Comment by rok-cesnovar** [comment]
This was further discussed at the Math meeting and the general feedback was that everyone present (Daniel, Ben and Marco) was in favor of something along these lines, but merge conflicts if going with a release-notes.md file would be problematic. And I agree. 

Wasting 7 hours of CI for a merge conflict on a .md file is bad. Opening a separate PR to just fix the .md file and skip CI tests is also bad. But both are still better than having to create 3 months worth of release notes.

The counter proposal was to have a field in either PRs or issues that would be easily collected with a script. This was already proposed once by @SteveBronder and was also discussed at length. I am not sure which one of those would be better. I slightly prefer PRs but any way of them would be better.

We can run the script on every merge to develop in the CI and publish the release notes on the wiki.  via the https://github.com/stan-dev/math.wiki.git repository. That way we have an up to date release notes at all times which is nice and whoever does the release doesnt have to run any script manually. 

We are probably going to move away from the wiki docs as proposed in https://github.com/stan-dev/math/issues/1658 but I think keeping up-to-date release notes can stay there.

**Comment by rok-cesnovar** [comment]
Since the idea behind this has changed to basically what Steve proposed in https://github.com/stan-dev/math/issues/1357 I am going to close this issue. We need to decide whether to use issues or PRs and then just do it.

**Comment by SteveBronder** [comment]
I like your idea more, than in the PR template we can just have a checkbox for "add a release notes line in release-notes.md if applicable"
